tsdb/wlog/wlog.go

```diff
@@ -244,78 +226,78 @@ func (w *wlMetrics) Unregister() {
+		}),
+		writesFailed: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Name: "writes_failed_total",
+			Help: "Total number of write log writes that failed.",
+		}),
+		walFileSize: promauto.With(r).NewGaugeFunc(prometheus.GaugeOpts{
+			Name: "storage_size_bytes",
+			Help: "Size of the write log directory.",
+		}, func() float64 {
+			val, err := w.Size()
+			if err != nil {
+				w.logger.Error("Failed to calculate size of \"wal\" dir", "err", err.Error())
+			}
+			return float64(val)
+		}),
+		recordPartWrites: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Name: "record_part_writes_total",
+			Help: "Total number of record parts written before flushing.",
+		}),
+		recordPartBytes: promauto.With(r).NewCounter(prometheus.CounterOpts{
+			Name: "record_parts_bytes_written_total",
+			Help: "Total number of record part bytes written before flushing, including" +
+				" CRC and compression headers.",
+		}),
+		recordBytesSaved: promauto.With(r).NewCounterVec(prometheus.CounterOpts{
+			Name: "record_bytes_saved_total",
+			Help: "Total number of bytes saved by the optional record compression." +
+				" Use this metric to learn about the effectiveness compression.",
+		}, []string{"compression"}),
 	}
-
-	m.fsyncDuration = prometheus.NewSummary(prometheus.SummaryOpts{
-		Name:       "fsync_duration_seconds",
-		Help:       "Duration of write log fsync.",
-		Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
-	})
-	m.pageFlushes = prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "page_flushes_total",
-		Help: "Total number of page flushes.",
-	})
-	m.pageCompletions = prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "completed_pages_total",
-		Help: "Total number of completed pages.",
-	})
-	m.truncateFail = prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "truncations_failed_total",
-		Help: "Total number of write log truncations that failed.",
-	})
-	m.truncateTotal = prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "truncations_total",
-		Help: "Total number of write log truncations attempted.",
-	})
-	m.currentSegment = prometheus.NewGauge(prometheus.GaugeOpts{
-		Name: "segment_current",
-		Help: "Write log segment index that TSDB is currently writing to.",
-	})
-	m.writesFailed = prometheus.NewCounter(prometheus.CounterOpts{
-		Name: "writes_failed_total",
-		Help: "Total number of write log writes that failed.",
-	})
-	m.walFileSize = prometheus.NewGaugeFunc(prometheus.GaugeOpts{
-		Name: "storage_size_bytes",
-		Help: "Size of the write log directory.",
-	}, func() float64 {
-		val, err := w.Size()
-		if err != nil {
-			w.logger.Error("Failed to calculate size of \"wal\" dir",
-				"err", err.Error())
-		}
-		return float64(val)
-	})
-
-	if r != nil {
-		r.MustRegister(
-			m.fsyncDuration,
-			m.pageFlushes,
-			m.pageCompletions,
-			m.truncateFail,
-			m.truncateTotal,
-			m.currentSegment,
-			m.writesFailed,
-			m.walFileSize,
-		)
-	}
-
-	return m
```

Comment:
I can't find where the metrics registration is happening now. Is that intentional?

Reply:
OH, completely missed it!